### PR TITLE
Change c_int to c_longlong for entry_set_size

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
While working with large files I found out that if entry is bigger than > 3.5 GB there is error in type- int is too small.

### CODE REVIEW

File: libarchive/ffi.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def ffi(name, argtypes, restype, errcheck=None):
 ffi('entry_rdevmajor', [c_archive_entry_p], c_uint)
 ffi('entry_rdevminor', [c_archive_entry_p], c_uint)
 
-ffi('entry_set_size', [c_archive_entry_p, c_int], c_int)
+ffi('entry_set_size', [c_archive_entry_p, c_longlong], c_longlong)
 ffi('entry_set_filetype', [c_archive_entry_p, c_int], c_int)
```
Comment (pombredanne) [inline libarchive/ffi.py]: Good catch! Thanks. Sounds like it has been like that for a long while. https://github.com/libarchive/libarchive/blob/7f17c791dcfd8c0416e2cd2485b19410e47ef126/libarchive/archive_entry.h#L244
Do you think there could be a test that could be written with some archive that fakes having a 3.5GB entry?

File: libarchive/ffi.py [hunk 2]
```diff
@@ -125,7 +125,7 @@ def ffi(name, argtypes, restype, errcheck=None):
 ffi('entry_rdevmajor', [c_archive_entry_p], c_uint)
 ffi('entry_rdevminor', [c_archive_entry_p], c_uint)
 
-ffi('entry_set_size', [c_archive_entry_p, c_int], c_int)
+ffi('entry_set_size', [c_archive_entry_p, c_longlong], c_longlong)
```
Comment (Changaco) [inline libarchive/ffi.py]: Why did you change the return type? `entry_set_size()` doesn't return anything.

File: libarchive/write.py [hunk 3]
```diff
@@ -98,7 +103,7 @@ def add_file_from_memory(
             archive_entry = ArchiveEntry(None, archive_entry_pointer)
 
             archive_entry.pathname = entry_path
-            entry_set_size(archive_entry_pointer, entry_size)
+            entry_set_size(archive_entry_pointer, long(entry_size))
```
Comment (Changaco) [inline libarchive/write.py]: What does that accomplish? Is it to coerce `int` to `long` when under python2?

Comment (krzysztofzuraw) [inline libarchive/write.py]: @Changaco I wanted to make sure that entry_size is always long type

File: libarchive/ffi.py (hunk 1 above)
Comment (krzysztofzuraw) [inline libarchive/ffi.py]: @pombredanne : I can write dummy 3.5 GB file to disk using simple for loop but I believe it will have bad impact on test performance.

**Comment by krzysztofzuraw** [comment]
@Changaco: could you rereview?

File: libarchive/ffi.py [hunk 4]
```diff
@@ -125,7 +125,7 @@ def ffi(name, argtypes, restype, errcheck=None):
 ffi('entry_rdevmajor', [c_archive_entry_p], c_uint)
 ffi('entry_rdevminor', [c_archive_entry_p], c_uint)
 
-ffi('entry_set_size', [c_archive_entry_p, c_int], c_int)
+ffi('entry_set_size', [c_archive_entry_p], c_longlong)
```
Comment (Changaco) [inline libarchive/ffi.py]: Unfortunately that's worse, it should be `ffi('entry_set_size', [c_archive_entry_p, c_longlong], c_int)`. The list represents the arguments passed to the C function. The fact that the tests didn't catch this mistake means that they're not very thorough.

Comment (krzysztofzuraw) [inline libarchive/ffi.py]: Ok, thank you- it is my first time I use ctypes module. What tests do you think will help to avoid such mistakes?

Comment (Changaco) [inline libarchive/ffi.py]: The way we test things in this library is mostly by creating archives and then reading them back. Comparing the two data sets allows us to detect bugs. For example, if you write an entry size of 42 and you read back a size of 0 or 10000, then you know that something's wrong.

**Comment by krzysztofzuraw** [comment]
@Changaco : I squashed the commit-  should I merge if all check are fine?
